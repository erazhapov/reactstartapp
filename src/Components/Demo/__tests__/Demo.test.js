import React from 'react';
import { shallow } from 'enzyme';
import Demo from '../Demo';

describe('Demo test', () => {
    it('Component props usage', () => {
        const demo = shallow(<Demo message="123" />);
        expect(demo.find('h1').text()).toEqual('123');
    });

    it('Component rendering (Snapshot)', () => {
        const demo = shallow(<Demo message="Hi!" />);
        expect(demo).toMatchSnapshot();
    })
});