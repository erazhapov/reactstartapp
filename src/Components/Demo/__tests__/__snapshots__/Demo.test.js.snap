// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Demo test Component rendering (Snapshot) 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <h1
    className="demo"
>
    Hi!
</h1>,
  "nodes": Array [
    <h1
      className="demo"
>
      Hi!
</h1>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 1,
      "_context": Object {},
      "_currentElement": <Demo
        message="Hi!"
/>,
      "_debugID": 3,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": Demo {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {
          "message": "Hi!",
        },
        "refs": Object {},
        "state": null,
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 2,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <h1
          className="demo"
>
          Hi!
</h1>,
        "_debugID": 4,
        "_renderedOutput": <h1
          className="demo"
>
          Hi!
</h1>,
      },
      "_renderedNodeType": 0,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <Demo
    message="Hi!"
/>,
}
`;
